packages/react/src/components/hero/Hero.tsx

```diff
@@ -0,0 +1,275 @@
+    cardChildIndex: -1,
+    backgroundChildIndex: -1,
+    components: [],
+  };
+
+  getChildrenAsArray(children).forEach((child, index) => {
+    const { componentName } = (child.type as FCWithName) || {};
+    childProps.components.push(child);
+    switch (componentName) {
+      case 'ImageContainer': {
+        childProps.imageChildIndex = index;
+        break;
+      }
+      case 'WideImage': {
+        childProps.wideImageChildIndex = index;
+        break;
+      }
+      case 'Card': {
+        childProps.cardChildIndex = index;
+        break;
+      }
+      case 'BackgroundImage': {
+        childProps.backgroundChildIndex = index;
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  });
+  return childProps;
+};
+
+const Card = ({
+  children,
+  centered,
+  className,
+  ...elementAttributes
+}: React.PropsWithChildren<HTMLElementAttributes & { centered?: boolean }>) => {
+  const classNameList = centered
+    ? classNames(styles.card, styles.centeredContent, className)
+    : classNames(styles.card, className);
+  return (
+    <div {...elementAttributes} className={classNameList}>
+      {children}
+    </div>
+  );
+};
+
+Card.componentName = 'Card';
+
+const ImageContainer = (props: ImgElementAttributes) => {
+  return (
+    /* eslint-disable-next-line jsx-a11y/alt-text */
+    <img className={styles.image} {...props} aria-hidden alt="" />
+  );
+};
+
+ImageContainer.componentName = 'ImageContainer';
+
+const WideImage = (props: ImgElementAttributes) => {
+  return <ImageContainer {...props} />;
+};
+
+WideImage.componentName = 'WideImage';
+
+const BackgroundImage = (props: ImgElementAttributes) => {
+  return <ImageContainer {...props} />;
+};
+BackgroundImage.componentName = 'BackgroundImage';
+
+export const Hero = ({ children, theme, koros, ...elementAttributes }: HeroProps) => {
+  const { components, imageChildIndex, backgroundChildIndex, wideImageChildIndex, cardChildIndex } = pickChildProps(
+    children,
+  );
+  const editableTheme = { ...theme };
+  if (!editableTheme['--koros-color']) {
+    editableTheme['--koros-color'] = 'var(--background-color)';
+  }
+  const customThemeClass = useTheme<HeroCustomTheme>(styles.hero, editableTheme);
+  const korosStyle = { fill: 'var(--koros-color)' };
+  const hideKoros = !!koros?.hide;
+  const canKorosBeFlipped = koros?.flipHorizontal !== false;
+
+  const getHeroType = () => {
+    if (wideImageChildIndex > -1) {
+      return 'wideImage';
+    }
+    if (backgroundChildIndex > -1) {
+      // if background is first, then the Hero version is the one where card is floating over background image
+      // if background is not first, then the Hero version is the one with angled Koros
+      return backgroundChildIndex === 0 ? 'backgroundTop' : 'angledKoros';
+    }
+    if (imageChildIndex === -1) {
+      return 'textOnly';
+    }
+    return imageChildIndex === 0 ? 'imageLeft' : 'imageRight';
+  };
+
+  const type = getHeroType();
```

Comment:
If `variant` prop is used, I think it would remove the need for figuring out where the elements are, right? This `getHeroType` function is sort of hard to read so it would improve this bit here in my opinion.

Reply:
Removed.